carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/core/capability/CapabilitiesBuilder.java

```diff
@@ -0,0 +1,113 @@
+package com.qaprosoft.carina.core.foundation.webdriver.core.capability;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.MutableCapabilities;
+
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.ChromeCapabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.EdgeCapabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.FirefoxCapabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.GeckoCapabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.Mac2Capabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.SafariCapabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.UIAutomator2Capabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.WindowsCapabilities;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.impl.XCUITestCapabilities;
+
+public class CapabilitiesBuilder {
+
+    private Capabilities customCapabilities = null;
+    private String testName = StringUtils.EMPTY;
+    private OptionsType optionsType;
+
+    private CapabilitiesBuilder() {
+    }
+
+    public static CapabilitiesBuilder builder() {
+        return new CapabilitiesBuilder();
+    }
+
+    public CapabilitiesBuilder testName(String testName) {
+        this.testName = testName;
+        return this;
+    }
+
+    public CapabilitiesBuilder fromCustomCapabilities(Capabilities capabilities) {
+        this.customCapabilities = capabilities;
+        return this;
+    }
+
+    public CapabilitiesBuilder chooseOptionsType(OptionsType optionsType) {
+        this.optionsType = optionsType;
+        return this;
+    }
+
+    public Capabilities build() {
+        MutableCapabilities capabilities = null;
+
+        if (this.optionsType == null) {
+            throw new RuntimeException("Options type should be chosen");
+        }
+
+        switch (optionsType) {
+        case CHROME:
```

Comment:
any way to simplify switch and if inside?

Reply:
This class removed